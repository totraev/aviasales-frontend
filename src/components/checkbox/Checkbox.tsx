import React, { SFC } from 'react';
import styles from './Checkbox.module.css';

import { stopsMessage } from '../../helpers/stops';
import { ICheckboxProps } from './Checkbox.types';

const Checkbox: SFC<ICheckboxProps> = ({
  id,
  stops,
  value,
  onClick,
  onUncheckOther,
}) => (
  <div className={styles.wrap}>
    <label className={styles.label} htmlFor={id}>
      <input
        id={id}
        className={styles.checkbox}
        type="checkbox"
        checked={value}
        onClick={onClick}
      />
      <span className={styles.faceChecked}/>

      {stopsMessage(stops)}
    </label>

    <div className={styles.append}>
      <button
        className={styles.uncheckOther}
        onClick={onUncheckOther}
      >
        ТОЛЬКО
      </button>
    </div>
  </div>
);

export default Checkbox;
