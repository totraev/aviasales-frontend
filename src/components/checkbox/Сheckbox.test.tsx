import React from 'react'
import renderer from 'react-test-renderer'

import Checkbox from './Checkbox'

describe('Checkbox component', () => {
  it('Should renders correctly', () => {
    const tree = renderer
      .create(<Checkbox id="test" stops={1} value={true} onClick={null} onUncheckOther={null} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
