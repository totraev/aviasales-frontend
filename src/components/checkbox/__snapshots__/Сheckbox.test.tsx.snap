// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Should renders correctly 1`] = `
<div
  className="wrap"
>
  <label
    className="label"
    htmlFor="test"
  >
    <input
      checked={true}
      className="checkbox"
      id="test"
      onClick={null}
      type="checkbox"
    />
    <span
      className="faceChecked"
    />
    1 ПЕРЕСАДКА
  </label>
  <div
    className="append"
  >
    <button
      className="uncheckOther"
      onClick={null}
    >
      ТОЛЬКО
    </button>
  </div>
</div>
`;
